### PR TITLE
Feat/loan settle

### PR DESCRIPTION
**Pull request by AinoraZ**
Untested (due to severe time constraints and needing to get this out ASAP) Implementation of settling loans.
Edit: Tests can be found here - https://github.com/TeamInterject/BillChop/pull/76

Loan settling is done via payments (/api/payments). 
Following data is needed for creating a payment:
  - payerId -- the user who is paying back
  - receiverId -- the user who is getting paid back
  - total -- the amount being paid back
  - groupContextId -- the group context where payer owes receiver.

Payments can only be created if there is an unsettled loan between the payer and receiver in the given group. 
Payments cannot exceed the amount owed by the payer to the receiver (no overpay).

You can generate the expected payments that need to be done and received for a user by calling api/payments/user/{userId}?groupId={groupId}. Optionally this can be constrained to a single group by passing the optional query parameter groupId.

For better understanding, I'm providing a sample (slightly modified) output from expected payments api call for user A (/user/{userAGuid}):
```json5
[
  { //A needs to pay back B 65.01 in group1
    "Id": "80017959-9575-4545-ad2b-d270942c0717",
    "Amount": 65.01,
    "CreationTime": "2020-12-05T15:17:35.2476196+01:00",
    "Payer": {
      "Id": "bc99c24f-1e75-4a61-ffcd-08d89899067d",
      "Name": "A",
      "Email": "A@gmail.com"
    },
    "Receiver": {
      "Id": "bbf88adb-4528-4516-ffcb-08d89899067d",
      "Name": "B",
      "Email": "B@gmail.com"
    },
    "GroupContextId": "#{group1-guid}"
  },
  { //C needs to pay back A 91.66 in group1
    "Id": "34bc506e-4e65-4a93-8530-73e763665eb8",
    "Amount": 91.66,
    "CreationTime": "2020-12-05T15:17:35.254361+01:00",
    "Payer": {
      "Id": "52ef0501-71b1-4fbd-ffcc-08d89899067d",
      "Name": "C",
      "Email": "C@gmail.com"
    },
    "Receiver": {
      "Id": "bc99c24f-1e75-4a61-ffcd-08d89899067d",
      "Name": "A",
      "Email": "A@gmail.com"
    },
    "GroupContextId": "#{group1-guid}"
  },
  { //A needs to pay back C 23.11 in group2
    "Id": "a787c44a-96a4-4a70-9961-e7856397134a",
    "Amount": 23.11,
    "CreationTime": "2020-12-05T15:17:35.265241+01:00",
    "Payer": {
      "Id": "bc99c24f-1e75-4a61-ffcd-08d89899067d",
      "Name": "A",
      "Email": "A@gmail.com"
    },
    "Receiver": {
      "Id": "52ef0501-71b1-4fbd-ffcc-08d89899067d",
      "Name": "C",
      "Email": "C@gmail.com"
    },
    "GroupContextId": "#{group2-guid}"
  }
]
```

### CODE REVIEW

File: billchop-be/billchop-be/Services/PaymentService.cs [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+                .Select(paymentGroup => 
+                    new GroupedTotals(paymentGroup.Key, paymentGroup.Sum(payment => payment.Amount))
+                ).ToList();
+
+            var totals = loaneeOwesTotals
+                .FullJoin(
+                    loaneePayedBackTotals,
+                    (owes) => owes.GroupContextId,
+                    (owes) => new GroupedTotals(owes.GroupContextId, owes.Total),
+                    (payedBack) => new GroupedTotals(payedBack.GroupContextId, payedBack.Total),
```
Comment (MartynasBas) [inline billchop-be/billchop-be/Services/PaymentService.cs]: Do a left join since only paying back without loaning shouldn't happen.

**Review by tmaciulis22 — APPROVED** [review]
Nice job 👏🏼

**Comment by AinoraZ** [comment]
[Task 12: Expose API to Get And Create "Loan Settles"](https://trello.com/c/7QYKaf1R/130-task-12-expose-api-to-get-and-create-loan-settles)

**Comment by AinoraZ** [comment]
[Task 12: Move total loan amount calculation to backend](https://trello.com/c/WJWqMzbP/129-task-12-move-total-loan-amount-calculation-to-backend)

**Comment by AinoraZ** [comment]
[Task 12: Add Settling Model To DB](https://trello.com/c/LWc6KJhc/127-task-12-add-settling-model-to-db)

**Pull request merged**